mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```diff
@@ -0,0 +1,442 @@
+
+
+        if (rec1 == null) return Optional.empty();
+
+        return Optional.of(manager.getClan(transaction,rec1.getAllyId()));
+    }
+
+    /**
+     * Returns a record of info
+     * @param transaction the tx
+     * @return the record
+     */
+    private ClansClanInfoRecord getInformation(Transaction transaction) {
+        ClansClanInfoRecord rec =  transaction.getProperty(DSLContext.class)
+                .fetchOne(CLANS_CLAN_INFO,CLANS_CLAN_INFO.CLAN_ID.eq(this.clanID));
+
+        assert rec != null : "nullity check failed for record get!";
+
+        return rec;
+    }
+
+    /**
+     * Adds kills to a clan
+     * @param transaction the tx
+     * @param toAdd how much to add
+     * @return accurate amount of kills post transaction
+     */
+    public int addKills(Transaction transaction, int toAdd) {
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanKills() + toAdd;
+
+        rec.setClanKills(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_KILLS);
+
+        this.clanKills = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Adds deaths to a gg.solarmc.loader.clans.Clan
+     * @param transaction the tx
+     * @param toAdd amount of deaths to add
+     * @return accurate count of deaths post transaction
+     */
+    public int addDeaths(Transaction transaction, int toAdd) {
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanDeaths() + toAdd;
+
+        rec.setClanDeaths(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_DEATHS);
+
+        this.clanDeaths = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Adds assists to a gg.solarmc.loader.clans.Clan.
+     * @param transaction tx
+     * @param toAdd the amount you want to add
+     * @return Accurate assists after transaction
+     */
+    public int addAssists(Transaction transaction, int toAdd) {
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanAssists() + toAdd;
+
+        rec.setClanAssists(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_ASSISTS);
+
+        this.clanAssists = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Gets all clan members accurately
+     * @param transaction the tx
+     * @return the clan members
+     */
+    public Set<ClanMember> getClanMembers(Transaction transaction) {
+        Set<ClanMember> bruh = transaction.getProperty(DSLContext.class)
+                .select(CLANS_CLAN_MEMBERSHIP.USER_ID)
+                .from(CLANS_CLAN_MEMBERSHIP)
+                .where(CLANS_CLAN_MEMBERSHIP.CLAN_ID.eq(this.clanID))
+                .fetchSet((rec) -> new ClanMember(this.clanID,rec.value1(),manager));
+
+        this.members = bruh;
+
+        return bruh;
+    }
+
+    /**
+     * Adds a member to the clan. Note that this does not have a built in limit, API implementer needs to implement
+     * limit if we desire a limit on members
```

Comment:
It may be useful to have a `public int getClanSize(Transaction transaction)` method. What do you think?

Reply:
currentMembers.size();

Reply:
currentMembers().size() introduces the additional computation of retrieving the member objects.

I guess we can add `int getClanSize(Transaction transaction)` if we ever need to get the size of the members without getting the members themselves.